File: lab 2/lab2.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.InputMismatchException;

class lab2 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        
        int X = 0;
        int Y = 0;

        // Get input for X
        while (true) {
            try {
                System.out.print("Enter a value for X: ");
                X = input.nextInt();
                if (X >= 0) {
                    break; // Break out of the loop if input is a valid non-negative integer
                } else {
                    System.out.println("Invalid input. Please enter a non-negative integer.");
                }
            } catch (InputMismatchException e) {
                System.out.println("Invalid input. Please enter an integer.");
                input.nextLine(); // Consume the invalid input
            }
        }

        // Get input for Y
        while (true) {
            try {
                System.out.print("Enter a value for Y: ");
                Y = input.nextInt();
                if (Y >= 0) {
                    break; // Break out of the loop if input is a valid non-negative integer
                } else {
                    System.out.println("Invalid input. Please enter a non-negative integer.");
                }
            } catch (InputMismatchException e) {
                System.out.println("Invalid input. Please enter an integer.");
                input.nextLine(); // Consume the invalid input
            }
        }


        input.close();

        
        ArrayList<ArrayList> list = new ArrayList<>();
        for(int count = X; count > 0; count--){
            ArrayList<Integer> subList = new ArrayList<>();
            for(int i = 0; i < Y; i++){
                int value = count + (X*i);
                subList.add(value);
            }
            list.add(subList);
        }

        for (int i = 0; i < list.size(); i++) {
            System.out.println(list.get(i));
          }
    
    }
}

File: lab 2/lab2.erl
% 
% Lab 2
%

-module(lab2).
-author('Jason').

%
% -- Public --
%

-export([start/0]).

start() -> 
    X = read_integer("Enter a value for X: "),
    Y = read_integer("Enter a value for Y: "),
    Const = X,
    Flist = listLoop(X, Y, Const),
    io:format("~p~n", [Flist]).



%
% -- Private -- 
%

read_integer(Prompt) ->
    case io:fread(Prompt, "~d") of
        {ok, [Value]} when Value >= 0 ->
            Value;
        _ ->
            io:format("Invalid input. Please enter an integer.~n"),
            read_integer(Prompt)
    end.


listLoop(0, Y, Const) -> [];
listLoop(X, Y, Const) -> [elementLoop(Const, Y, X) | listLoop(X-1, Y, Const)].


elementLoop(Const, 1, X) ->[X];
elementLoop(Const, Y, X) -> Val = X + (Const*(Y-1)),
    [Val | elementLoop(Const, Y-1, X)].